controllers/statuscheck/controller.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package statuscheck
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/controllers/utils/recorder"
+)
+
+type Reconciler struct {
+	logger        logr.Logger
+	kubeClient    client.Client
+	eventRecorder recorder.ChaosRecorder
+	manager       Manager
+}
+
+func NewReconciler(logger logr.Logger, kubeClient client.Client, eventRecorder recorder.ChaosRecorder, manager Manager) *Reconciler {
+	return &Reconciler{
+		kubeClient:    kubeClient,
+		eventRecorder: eventRecorder,
+		logger:        logger,
+		manager:       manager,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	startTime := time.Now()
+	obj := &v1alpha1.StatusCheck{}
+	if err := r.kubeClient.Get(ctx, req.NamespacedName, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			r.logger.Info("status check is deleted")
+			// the StatusCheck is deleted, remove it from manger
+			r.manager.Delete(req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// if status check was completed previously, we don't want to redo the termination
+	if obj.IsCompleted() {
+		r.logger.Info("status check is already completed")
+		return ctrl.Result{}, nil
+	}
+
+	result, ok := r.manager.Get(*obj)
+	if !ok {
+		// if nil, add status check to manager
+		err := r.manager.Add(*obj)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
```

Comment:
Please notice that we do not resolve the changing/updating of spec, maybe we should also reject the updates of StatusCheck, like `var ErrCanNotUpdateChaos = errors.New("Cannot update chaos spec")`

Reply:
Yes, the statuscheck spec cannot be updated: https://github.com/iguoyr/chaos-mesh/blob/status-check-reconciler/api/v1alpha1/zz_generated.chaosmesh.go#L1807